lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -427,4 +401,65 @@ export class BrowserStorage extends CacheManager {
             throw BrowserAuthError.createTokenRequestCacheError(err);
         }
     }
+
+    /**
+     * helper to retrieve the value with type from the cache
+     * @param key
+     * @param value
+     */
+    private getItemHelper(key: string, value: string): ValidCacheType {
+
+        // handle null cases
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+
+        let parsedValue;
+
+        // handle strings - TEMPORARY Values; check cookies for these
+        try {
+            parsedValue = JSON.parse(value);
+        } catch (e) {
+            if (this.cacheConfig.storeAuthStateInCookie) {
+                const itemCookie = this.getItemCookie(key);
+                return itemCookie;
+            }
+            return value;
+        }
+
+        // Account
+        if (parsedValue.hasOwnProperty(AUTHORITY_TYPE)) {
+            const accountEntity = new AccountEntity();
+            return CacheManager.toObject(accountEntity, parsedValue) as AccountEntity;
```

Comment:
All of this casting business logic should be pulled into helper functions one level up (i.e. functions that call `getItem`).